src/Pages/ADT3DScenePage/ADT3DScenePage.tsx

```diff
@@ -209,27 +224,38 @@ const ADT3DScenePageBase: React.FC<IADT3DScenePageProps> = ({
         [environmentPickerOptions?.storage, setBlobContainerUrl]
     );
 
-    const handleEnvironmentUrlChange = useCallback(
-        (envUrl: string, envUrls: Array<string>) => {
+    const handleAdtInstanceChange = useCallback(
+        (
+            adtInstance: string | IADTInstance,
+            adtInstances: Array<string | IADTInstance>
+        ) => {
             deeplinkDispatch({
-                type: DeeplinkContextActionType.SET_ADT_URL,
+                type: DeeplinkContextActionType.SET_ADT_INSTANCE,
                 payload: {
-                    url: envUrl
+                    adtInstance
                 }
             });
-            if (environmentPickerOptions?.environment?.onEnvironmentChange) {
-                environmentPickerOptions.environment.onEnvironmentChange(
-                    envUrl,
-                    envUrls
+            if (environmentPickerOptions?.adt?.onAdtInstanceChange) {
+                environmentPickerOptions.adt.onAdtInstanceChange(
+                    adtInstance,
+                    adtInstances
                 );
             }
         },
-        [deeplinkDispatch, environmentPickerOptions?.environment]
+        [deeplinkDispatch, environmentPickerOptions?.adt]
     );
 
     // update the adapter if the ADT instance changes
     useEffect(() => {
         adapter.setAdtHostUrl(deeplinkState.adtUrl);
```

Comment:
The only difference between updating the adapter values for ADT instance and blob storage URL based on the changes in deeplink is the additional part in ADT instance to update the selected adt url in localstorage for the consumer side (as I mentioned in my other comment, it was for whenever 3d scene studio app updates the selected adt url, ADT explorer app should also do that - this is not being done automatically now since we moved towards other local storage key structure, that is why I needed to do it manually here, for now.). The reason why I moved it from callback to useeffect that the consumer localstorage was **not** being updated on page load when the deeplink adt url was from parsed, since the callback has not being called yet due to no interaction with the EnvironmentPicker. 

Does it make sense?

Reply:
Gotcha that makes sense if the callback never triggers. I'm just thinking it would be safer to handle both the same way. Is there a downside to updating the blobUrl using a useEffect as well?

Reply:
the containerUrl is already being updated in useEffect as well? 
<img width="749" alt="image" src="https://user-images.githubusercontent.com/45217314/193899034-d0fdd915-986a-4665-a275-5f2d0f40ddc1.png">